crates/dslab-dag/src/parsers/dot_parser.rs

```diff
@@ -0,0 +1,176 @@
+            continue;
+        }
+        let value = &value[1..value.len() - 1];
+        result.insert(name.iter().collect(), value.iter().collect());
+    }
+    result
+}
+
+impl DAG {
+    pub fn from_dot(file: &str) -> Self {
+        let lines = std::fs::read_to_string(file)
+            .expect(&format!("Can't read file {}", file))
+            .trim()
+            .split('\n')
+            .map(|x| x.to_string())
+            .collect::<Vec<String>>();
+
+        let is_word_char = |c: char| -> bool { c.is_ascii_alphanumeric() || c == '_' };
+
+        let mut nodes: Vec<Node> = Vec::new();
+        let mut edges: Vec<Edge> = Vec::new();
+
+        for line in lines.into_iter() {
+            let line = line.trim();
+            if line.is_empty() {
+                continue;
+            }
+
+            let chars = line.chars().collect::<Vec<_>>();
+            let mut i = 0;
+            while i < chars.len() && is_word_char(chars[i]) {
+                i += 1;
+            }
+            if i == 0 {
+                continue;
+            }
+            let token1 = &chars[0..i];
+            if token1.iter().collect::<String>() == "edge" || token1.iter().collect::<String>() == "node" {
+                // reserved special name
+                continue;
+            }
+            while i < chars.len() && chars[i].is_ascii_whitespace() {
+                i += 1;
+            }
+            if i == chars.len() {
+                continue;
+            }
+            if chars[i] == '-' {
+                // edge
+                // token1 -> token2 [label="1->2",size="1.5"]
+                i += 1;
+                if i == chars.len() || chars[i] != '>' {
+                    continue;
+                }
+                i += 1;
+                while i < chars.len() && chars[i].is_ascii_whitespace() {
+                    i += 1;
+                }
+                let token_start = i;
+                while i < chars.len() && is_word_char(chars[i]) {
+                    i += 1;
+                }
+                if token_start == i {
+                    continue;
+                }
+                let token2 = &chars[token_start..i];
+                while i < chars.len() && chars[i].is_ascii_whitespace() {
+                    i += 1;
+                }
+                if i == chars.len() || chars[i] != '[' {
+                    continue;
+                }
+                let mut r = chars.len() - 1;
+                while r > i && chars[r] != ']' {
+                    r -= 1;
+                }
+                if r <= i {
+                    continue;
+                }
+                let mut params = parse_params(&chars[i + 1..r]);
+                edges.push(Edge {
+                    from: token1.iter().collect(),
+                    to: token2.iter().collect(),
+                    label: params.remove("label"),
+                    size: params.get("size").map(|x| x.parse::<f64>().unwrap_or(0.)).unwrap_or(0.),
```

Comment:
Предлагаю здесь и ниже не делать `unwrap_or`, маскируя ошибки (что может запутать пользователя), а просто падать, если с `size` что-то не так:

```suggestion
                    size: params.get("size").unwrap().parse::<f64>().unwrap(),
```